internal/promotion/runner/builtin/git_cloner.go

```diff
@@ -103,10 +103,24 @@ func (g *gitCloner) run(
 			SSHPrivateKey: creds.SSHPrivateKey,
 		}
 	}
+	var repoUser git.User
+
+	if cfg.User != nil {
+		repoUser = git.User{
+			Name:  cfg.User.Name,
+			Email: cfg.User.Email,
+			SigningKey:     cfg.User.SigningKey, // Optional, may be empty
+			SigningKeyType: g.gitUser.SigningKeyType, // Preserve system-level SigningKeyType
+			SigningKeyPath: g.gitUser.SigningKeyPath, // Preserve system-level SigningKeyPath
```

Comment:
If `SigningKey` is non-empty, the `git` package, internally, will write it to a file and use its path as the `SigningKeyPath`, so there's no need to specify `SigningKeyPath` here. In fact, doing so with it's value set to something system-level seems likely to promote confusion. Let's just remove this line.

```suggestion
```